zero_bin/leader/src/main.rs

```diff
@@ -36,6 +37,10 @@ fn get_previous_proof(path: Option<PathBuf>) -> Result<Option<GeneratedBlockProo
     Ok(Some(proof))
 }
 
+const SEGMENT_PROOF_ROUTING_KEY: &str = "segment_proof";
+const BLOCK_PROOF_ROUTING_KEY: &str = "block_proof";
+const DEFAULT_ROUTING_KEY: &str = paladin::runtime::DEFAULT_ROUTING_KEY;
```

Comment:
These are the only "modes" we need?

Reply:
Yeah I know they're not actually modes, but they also sort of are in that they're dictating where things are routed in a given situation. I can rephrase to, are these the only keys we care about

Reply:
Yes

Reply:
I guess I'm a bit confused by the naming @temaniarpit27. Is `BLOCK_PROOF_ROUTING_KEY` used for both `SegmentAggProof` / `BatchAggProof` and `BlockProof` or only the latter? 
Quid of `SEGMENT_PROOF_ROUTING_KEY`?

We need to decouple `SegmentProof` (many occurences, heavy processes) to the three other kind of proofs (`SegmentAggProof` / `BatchAggProof` / `BlockProof`) which are less numerous and fairly light in comparison (orders of magnitude).

cc @leovct

Reply:
so basically i was sending segmentproof, segmentaggproof to segment-proof machines and batchproof and blockproof to block-proof machines

i will separate segmentproof to rest 3

Reply:
Given this change, it would make sense to think of better names to group these proofs, maybe `light-proof` and `heavy-proof` instead of `block-proof` and `segment-proof`?

Reply:
i am also up for better naming

Reply:
Any other suggestions?

Reply:
I'm good with Leo's suggestion, it carries the meaning behind this optimization.